src/docker_clojure/core.clj

```diff
@@ -14,15 +14,19 @@
    [docker-clojure.log :refer [log] :as logger]
    [clojure.edn :as edn]))
 
-(defn contains-every-key-value?
-  "Returns true if the map `haystack` contains every key-value pair in the map
-  `needles`. `haystack` may contain additional keys that are not in `needles`.
-  Returns false if any of the keys in `needles` are missing from `haystack` or
-  have different values."
-  [haystack needles]
+(defn matches?
+  "Returns true if the map `variant` contains every key-value pair in the map
+  `exclusion`. `variant` may contain additional keys that are not in
+  `exclusion`. Some values of `exclusion` can also be a predicate of one
+  argument which is then tested against the respective value from `variant`.
+  Returns false if any of the keys in `exclusions` are missing from `variant` or
+  have different values, or the predicate value returned false."
+  [variant exclusion]
   (every? (fn [[k v]]
-            (= v (get haystack k)))
-          needles))
+            (if (fn? v)
+              (v (get variant k))
+              (= v (get variant k))))
+          exclusion))
```

Comment:
I think now that this fn is pretty specific to checking exclusions against variants, perhaps a more specific name is in order? Maybe something like `exclude-variant?`